pkg/controller/csr_check.go

```diff
@@ -131,20 +133,30 @@ func authorizeCSR(
 	req *certificatesv1.CertificateSigningRequest,
 	csr *x509.CertificateRequest,
 	ca *x509.CertPool,
-) error {
+) (bool, error) {
 	if req == nil || csr == nil {
-		return fmt.Errorf("Invalid request")
+		klog.Errorf("authorizeCSR invalid request")
+		return false, nil
 	}
 
 	if isNodeClientCert(req, csr) {
-		return authorizeNodeClientCSR(c, config, machines, req, csr)
+		if config.NodeClientCert.Disabled {
+			klog.Errorf("%v: CSR rejected as the flow is disabled", req.Name)
+			return false, fmt.Errorf("CSR %s for node client cert rejected as the flow is disabled", req.Name)
+		}
+		return authorizeNodeClientCSR(c, machines, req, csr)
 	}
 
+	klog.Infof("%v: CSR does not appear to be client csr", req.Name)
 	// node serving cert validation after this point
 
 	nodeAsking, err := validateCSRContents(req, csr)
-	if err != nil {
-		return err
+	if nodeAsking == "" {
```

Comment:
What happens to the `err` here? Why are we ignoring it? Will it always be nil if `nodeAsking` is populated? Can that be guaranteed into the future? Maybe check it explicitly as well?
```suggestion
	if nodeAsking == "" || err != nil {
```